doc/api/buffer.md

```diff
@@ -1300,9 +1300,9 @@ If `value` is not a string, number, or `Buffer`, this method will throw a
 `TypeError`. If `value` is a number, it will be coerced to a valid byte value,
 an integer between 0 and 255.
 
-If `byteOffset` is not a number, it will be coerced to a number. Any arguments
-that coerce to `NaN` or 0, like `{}`, `[]`, `null` or `undefined`, will search
-the whole buffer. This behavior matches [`String#indexOf()`].
+If `byteOffset` is not a number, it will be coerced to a number. If an argument
+is coerced to `NaN` or `0`, then the entire buffer will be searched. This
```

Comment:
Un the first part of the sentance, this looks a little less clear to me with the «coerced» word.

Perhaps, «For arguments that coerce to … the entire buffer will be searched» or something similar?

Reply:
`byteOffset` is the only argument here for which the statement is applicable so maybe this?

> If `byteOffset` is not a number, it will be coerced to a number. If it is
> coerced to `NaN` or `0`, then the entire buffer will be searched. This

...of this?:

> If `byteOffset` is not a number, it will be coerced to a number. If the result
> of coercion is `NaN` or `0`, then the entire buffer will be searched. This

Reply:
The second one looks perfect!